lms/djangoapps/course_home_api/course_metadata/v1/tests/test_views.py

```diff
@@ -77,3 +77,4 @@ def test_streak_data_in_response(self):
         response = self.client.get(self.url, content_type='application/json')
         celebrations = response.json()['celebrations']
         assert 'streak_length_to_celebrate' in celebrations
+        assert 'streak_discount_experiment_enabled' in celebrations
```

Comment:
updated the tests, do you think i should create a mixin for this test since the code is duplicated?  
or should I test the shared utility function and not check for the values here?

Reply:
I think what you did is fine. If you could only have one of "test shared utility function" and "test views", I'd rather have "test views" (as it is more on the functional side than unit side). Both are also possible, since these view tests are kind of happy-path, and maybe a unit test checking the various code paths of the uitility function would also be good?

As for a mixin, :shrug: duplication in test code doesn't bug me as much, especially for fringe stuff like this that will likely only show up a few times. But could do, if you think it makes things cleaner.